index.js

```
@@ -1,158 +1,195 @@
+ */
+async function getJoinedChats(sid, com) {
+    let threadList = objs.threadList;
+    if(typeof sid != 'string' || typeof com !== 'string') {
+        throw new Error('All Arguments are not satisfied.');
+    }
+    await request.get(endpoints.getJoinedChats(com), {
+        headers: {
+            'NDCAUTH': `sid=${sid}`
+        }
+    }, (err, res, body) => {
+        try {
+            //Parsing the Response.
+            body = JSON.parse(body);
+            body.threadList.forEach((element) => {
+                //Sorting the Elements and pushing them into the Array.
+                threadList.threads.push(sorter.threadSort(element));
+            });
+            threadList.status = 'ok';
+            threadList.error = null;
+        } catch (err) {
+            threadList.error = err;
+        }
+    });
+    return threadList;
+}
 
-    getJoinedComs: async function(sid) {
-        let communityList = objs.communityList;
-        communityList.coms = [];
-        await request.get(endpoints.getComs, {
+/**
+ * Loads Messages in a specific Chat.
+ * @param {SecurityString} sid For authenticating with the Narvii-API.
+ * @param {CommunityUUID} com The Community ID that can be Obtained by the function getJoinedComs
+ * @param {ChatUUID} uid The Chats ID that can be obtained by the function getJoinedChats
+ * @param {Number} count The ammount of Messages to Load (defaults to 1);
+ * @returns {Object} Object where all the Messages in the requested Chat are contained in an Array.
+ */
+async function getChat(sid, com, uid, count) {
+    let msgList = objs.recivedMessages;
+    if(typeof sid != 'string' || typeof com !== 'string' || typeof uid !== 'string') {
+        throw new Error('All Arguments are not satisfied.');
+    }
+    if(count == undefined || count == null) {
+        count = 1;
+        console.log('[AminoJS.getChat()] No count Variable Defined. Defaulting to 1'); // eslint-disable-line 

```

General Question. Should this Stay here?

---

A silent fallback is more favorable.

---

will do.